e2e/specs/hostframe.spec.js

```diff
@@ -15,118 +15,149 @@
  */
 
 describe("Host communication", () => {
-    before(() => {
-        cy.visit("hostframe.html");
-        cy.get("#toolbar button").should("have.length", 10);
-        cy.get("iframe")
-            .iframe(() => {
-                cy.waitForScriptFinish()
-            });
+  before(() => {
```

Comment:
Some things to note:

* The vast majority of the changes to this file were applied by the autoformatter. Not sure how this file
   ended up with 4-space indents instead of 2.
* The only actual change made to this file is the test added to the very end
* We should be migrating this test to playwright soon, but it only took ~5 minutes to add a test here
   vs probably the day+ of work it'll take to migrate this test, so I think it's still worth doing this now
   to get the PR merged sooner than later. I added a TODO for myself to go back and migrate things later.

Reply:
I actually had this same issue as well. I think we should migrate this to playwright, but I was going to suggest after MPA v2.

Reply:
@raethlein added a new playwright fixture to run an app in iframe recently. The hostframe logic could potentially be integrated into this iframe fixture.

Reply:
here is an example test using the new fixture: https://github.com/streamlit/streamlit/blob/develop/e2e_playwright/st_dataframe_interactions_test.py#L329 in case it helps 🙂

Reply:
Hm, I'll leave this as a TODO for now since I think there might be a bit more work to get things embedded correctly in `hostframe.html`, and I'd prefer to go ahead and get this merged to wrap up the quarterly goal cleanly. I'll most likely come back around to migrating this file in a few weeks time.